ecs-cli/modules/cli/compose/entity/entity_helper.go

```diff
@@ -134,8 +134,12 @@ func createRegisterTaskDefinitionRequest(taskDefinition *ecs.TaskDefinition, tag
 		PlacementConstraints:    taskDefinition.PlacementConstraints,
 	}
 
-	// 2023-11 Unconditionally set somaxconns at the task ContainerDefinition level.
-	if len(taskDefinition.ContainerDefinitions) > 0 {
+	if networkMode := taskDefinition.NetworkMode; aws.StringValue(networkMode) != "" {
+		request.NetworkMode = networkMode
+	}
+
+	// 2023-11 Conditionally set somaxconns at the task ContainerDefinition level.
+	if len(taskDefinition.ContainerDefinitions)  > 0 && aws.StringValue(taskDefinition.NetworkMode)  == "awsvpc" {
```

Comment:
For folks reviewing, `aws.StringValue` does the following:
```
// StringValue returns the value of the string pointer passed in or
// "" if the pointer is nil.
```

taskDefinition.NetworkMode is a string pointer, so this is exactly what we need!